GitUI/CommandsDialogs/BrowseDialog/DashboardControl/Dashboard.cs

```diff
@@ -372,7 +177,11 @@ private void groupLayoutPanel_DragDrop(object sender, DragEventArgs e)
                         }
                     }
 
-                    Repositories.AddMostRecentRepository(module.WorkingDir);
+                    ThreadHelper.JoinableTaskFactory.RunAsync(async () =>
+                    {
+                        await TaskScheduler.Default.SwitchTo(alwaysYield: true);
```

Comment:
💡  The explicit `SwitchTo` here adds unnecessary complexity (since the caller is on the UI thread), and should probably be removed.

Reply:
I am very much unclear as to when `.SwitchTo(alwaysYield: true)` should or shouldn't be called.
I've looked at [docs for AwaitExtensions.SwitchTo](https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.threading.awaitextensions.switchto?redirectedfrom=MSDN&view=visualstudiosdk-2017) but I am still confused...
I've looked at other instances of use in the codebase and it seemed to be consistent. E.g. 
https://github.com/gitextensions/gitextensions/blob/80bbf4dc9411ea0772962c96bfc4e41b85a11186/GitUI/CommitInfo/CommitInfo.cs#L213
https://github.com/gitextensions/gitextensions/blob/80bbf4dc9411ea0772962c96bfc4e41b85a11186/GitUI/CommitInfo/CommitInfo.cs#L236-L239

It would appreciated, if you could explain.

Reply:
> I am very much unclear as to when .SwitchTo(alwaysYield: true) should or shouldn't be called.

It should typically not be called in new code. I used it during the initial conversion to preserve behavior of the original code as a regression prevention strategy. During the ongoing migration to asynchronous code, it may be necessary to insert an explicit-yielding-switch if the following action synchronously executes long-running operations (e.g. a blocking call to Git rather than an asynchronous call) and the caller must not block on this operation yet might already be on a background thread.